pkg/api/v1beta3/types.go

```diff
@@ -170,12 +170,14 @@ type Volume struct {
 // VolumeSource represents the source location of a valume to mount.
 // Only one of its members may be specified.
 type VolumeSource struct {
-	// HostDir represents a pre-existing directory on the host machine that is directly
-	// exposed to the container. This is generally used for system agents or other privileged
-	// things that are allowed to see the host machine. Most containers will NOT need this.
+	// HostPath represents a pre-existing file or directory on the host
+	// machine that is directly exposed to the container. This is generally
+	// used for system agents or other privileged things that are allowed
+	// to see the host machine. Most containers will NOT need this.
 	// TODO(jonesdl) We need to restrict who can use host directory mounts and who can/can not
 	// mount host directories as read/write.
-	HostDir *HostDir `json:"hostDir"`
+	HostPath *HostPath `json:"hostPath"`
+	HostDir  *HostPath `json:"hostDir"` // DEPRECATED: use HostPath instead
```

Comment:
It's a minor point, but I have been asked a few times whether HostDir
supported files, too.  It does -> rename.  I'll drop the compat stuff if
you think compat is not so compelling (though I find the cost to be minor).

Say yes one more time, and I will do it :)

On Tue, Jan 20, 2015 at 3:08 PM, Brian Grant <notifications@github.com>
wrote:

> In pkg/api/v1beta3/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3517#discussion_r23265113>
> :
>
> >  	// TODO(jonesdl) We need to restrict who can use host directory mounts and who can/can not
> >  	// mount host directories as read/write.
> > -	HostDir *HostDir `json:"hostDir"`
> > +	HostPath *HostPath `json:"hostPath"`
> > +	HostDir  *HostPath `json:"hostDir"` // DEPRECATED: use HostPath instead
>
> If we believe the change is compelling, let's just do it. If not, let's
> leave it as is. Most users shouldn't be using this field, anyway.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3517/files#r23265113>
> .
>